core/src/main/java/org/apache/accumulo/core/conf/AccumuloConfiguration.java

```diff
@@ -209,13 +209,19 @@ static public long getMemoryInBytes(String str) {
         case 'B':
           multiplier = 0;
           break;
+        case '%':
+          int percent = Integer.parseInt(str.substring(0, str.length() - 1));
+          if (percent <= 0 || percent >= 100) {
+           throw new IllegalArgumentException("The value of '" + str + "' is not a valid memory percentage setting.");
+          }
+          return Runtime.getRuntime().maxMemory() * percent / 100;
         default:
           return Long.parseLong(str);
       }
       return Long.parseLong(str.substring(0, str.length() - 1)) << multiplier;
     } catch (Exception ex) {
       throw new IllegalArgumentException("The value '" + str + "' is not a valid memory setting. A valid value would a number "
-          + "possibily followed by an optional 'G', 'M', 'K', or 'B'.");
+          + "possibly followed by an optional 'G', 'M', 'K', or 'B'.");
```

Comment:
Since this is catching exception, it possible that something could go wrong parsing a `%`.  So error message should include `%` as valid suffix.